src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -762,8 +888,46 @@ internal override void Complete()
                     }
 
                     this.Command.Context.SetVariable(SpecialVariables.LastExitCodeVarPath, _nativeProcess.ExitCode);
-                    if (_nativeProcess.ExitCode != 0)
-                        this.commandRuntime.PipelineProcessor.ExecutionFailed = true;
+                    if (_nativeProcess.ExitCode == 0)
+                    {
+                        return;
+                    }
+
+                    this.commandRuntime.PipelineProcessor.ExecutionFailed = true;
+
+                    if (!ExperimentalFeature.IsEnabled(ExperimentalFeature.PSNativeCommandErrorActionPreferenceFeatureName)
+                        || !Command.Context.GetBooleanPreference(
+                                SpecialVariables.PSNativeCommandUseErrorActionPreferenceVarPath,
+                                InitialSessionState.DefaultPSNativeCommandUseErrorActionPreference,
```

Comment:
It might cover the conversion, but without declaring the converter, users would be able to set arbitrary values to the preference variable, wouldn't they? Better to keep its type consistent, IMO, less confusion for users.

Reply:
That certainly makes sense. If you want to keep the converter attribute, then I think there is no need to use `GetBooleanPreference`.

Reply:
What's the alternative API to use here? We'll still need to get a boolean conversion here to use it in the `if` condition, right?

Reply:
How about just casting the value to bool, since we know it has to be a bool.

Reply:
Used `GetVariableValue` and an explicit cast